scripts/bert/create_pretraining_data.py

```diff
@@ -0,0 +1,464 @@
+# coding=utf-8
+# Copyright 2018 The Google AI Language Team Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Create masked LM/next sentence masked_lm Gluon examples for BERT."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import logging
+import pickle
+import io
+import glob
+import collections
+import random
+import mxnet as mx
+import tokenizer as tokenization
+
+logging.getLogger().setLevel(logging.DEBUG)
+
+parser = argparse.ArgumentParser(
+    description='Pre-training data generator for BERT')
+
+parser.add_argument('--input_file', type=str, default=None, help='input_file')
+
+parser.add_argument(
+    '--output_file',
+    type=str,
+    default=None,
+    help='Output MXRecordIO example file (or comma-separated list of files).')
+
+parser.add_argument(
+    '--vocab_file',
+    type=str,
+    default=None,
+    help='The vocabulary file that the BERT model was trained on.')
+
+parser.add_argument(
+    '--do_lower_case',
+    action='store_true',
+    help='Whether to lower case the input text. Should be True for uncased'
+    'models and False for cased models.')
+
+parser.add_argument(
+    '--max_seq_length', type=int, default=128, help='Maximum sequence length.')
+
+parser.add_argument(
+    '--max_predictions_per_seq',
+    type=int,
+    default=20,
+    help='Maximum number of masked LM predictions per sequence.')
+
+parser.add_argument(
+    '--random_seed',
+    type=int,
+    default=12345,
+    help='Random seed for data generation.')
+
+parser.add_argument(
+    '--dupe_factor',
+    type=int,
+    default=10,
+    help='Number of times to duplicate the input data (with different masks).')
+
+parser.add_argument(
+    '--masked_lm_prob',
+    type=float,
+    default=0.15,
+    help='Masked LM probability.')
+
+parser.add_argument(
+    '--short_seq_prob',
+    type=float,
+    default=0.1,
+    help='Probability of creating sequences which are shorter than the '
+    'maximum length.')
+
+args = parser.parse_args()
+
+logging.info(args)
+
+
+class TrainingInstance:
```

Comment:
inherit from object for py2 compatibility

Reply:
Sure!